### ISSUE TITLE
RTStruct to NIFTI generates missing slices for non-constant z-axis

### ISSUE THREAD

**Issue by MLK97** [pyplati/platipy#165]
I am working on a set of MR images with RTStructs.
The protocol for generating those MR images is in such a way that the patient does not lie parallel but is slightly rotated in the z-axis (of the RTStruct). 
I was told this is in order to have the actual organ of interest parallel to the z-axis.
So therefore there is a difference between the patient z-axis and that RTStruct z-axis.

When converting this RTStruct to Nifti or a mask with either
```platipy.dicom.io.rtstruct_to_nifti.convert_rtstruct```
or
```platipy.dicom.io.rtstruct_to_nifti.transform_point_set_from_dicom_struct```
this leads to missing slices/holes in the generated mask.

![Rectum Nifti Mask](https://user-images.githubusercontent.com/18293577/200847785-0baa3e68-0fea-4367-a6bb-bdebf5eadc82.png)

I assume this might be caused from the assumption during the mask generation that the z-index is constant inside each closed contour defined in the RTStruct.

https://github.com/pyplati/platipy/blob/3432ae613cb893c1fefc2ab8b0e23243e602952e/platipy/dicom/io/rtstruct_to_nifti.py#L182-L188

I'll try to check if this behavior might be avoided if one removes this generalization but would also be happy to hear your thoughts about this.

**Comment by pchlap** [comment]
Hi @MLK97, thanks for the bug report.

Yes I believe you are correct in the assumption that this is caused by the z-index not being constant per slice. One option to fix this would be to rotate the contour points to an orientation where the z-index is constant. Then the resulting mask could be rotated back to the image space.

I don't have any data to be able to investigate this. If you happen to have an MRI image along with an RTSTRUCT file of a phantom or similar and you'd be happy to share this I could take a look. If not I can have a chat to our MR Radiographers to see if someone at our centre could scan a phantom with this kind of orientation (not sure if/when that would be possible though).

**Comment by MLK97** [comment]
Hey @pchlap,

sorry for the late reply, we did some further investigation into the dataset to make sure that the problem is really caused by this rotation problem and it surprisingly turned out that there instead were a lot of missing contour slices from the manual delineation process itself.
So while we are sure that for a lot of the studies, we looked at, the actual error stems from the manual delineation process and not from the rotation, it might as well be that there are studies that have an additional error that is caused by the rotation of the scan.
I will do some further control on my dataset and compare it to a Matlab package that has a rotation matrix built-in and get back to you in case I find anything.
If I don't find anything, I would close this issue unless you think it is worthwhile keeping this visible in case of other cases where this might become an issue in the conversion process.
